python/tank/util/storage_roots.py

```diff
@@ -145,7 +145,7 @@ def write(cls, sg_connection, config_folder, storage_roots):
         """
 
         (local_storage_lookup, unmapped_roots) = storage_roots.get_local_storages(
-            sg_connection
+            sg_connection, project_id
```

Comment:
I am not crazy about the new parameter that we take along everywhere. Is there no way to retrieve this information in another way?

Like I know tk-core provides a function called `current_engine()`. Is there the same for current project?

Reply:
We should also move the change more "upstream."
By the time we get here to write the file, we should already have a path that has been resolved relative to its' project.